File: src/main/java/com/kishlaly/ta/analyze/testing/sl/StopLossVolatileBollingerMiddle.java
package com.kishlaly.ta.analyze.testing.sl;

import com.kishlaly.ta.model.SymbolData;
import com.kishlaly.ta.model.indicators.BollingerJava;
import com.kishlaly.ta.utils.IndicatorUtils;

public class StopLossVolatileBollingerMiddle extends StopLossStrategy {

    public StopLossVolatileBollingerMiddle() {
        super(null, true);
    }

    @Override
    public double calculate(SymbolData data, int signalIndex) {
        BollingerJava bollinger = IndicatorUtils.buildBollingerBands(data.symbol, data.quotes).get(signalIndex);
        return bollinger.getMiddle();
    }

    @Override
    public String toString() {
        return "SL volatile Bollinger middle";
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/testing/sl/StopLossVolatileBollingerMiddle.kt
package com.kishlaly.ta.analyze.testing.sl

import com.kishlaly.ta.model.SymbolData
import com.kishlaly.ta.utils.IndicatorUtils

class StopLossVolatileBollingerMiddle : StopLossStrategy(isVolatile = true) {

    override fun calculate(data: SymbolData, signalIndex: Int): Double {
        return IndicatorUtils.buildBollingerBands(data.symbol, data.quotes)[signalIndex].middle
    }

    override fun toString() = "SL volatile Bollinger middle"
}